### ISSUE TITLE
subordinate charms have "num_units" added to bundle.yaml, fails demo deployment

### ISSUE THREAD

**Issue by kwmonroe** [canonical/jujucharms.com#100]
We have a subordinate charm called "plugin" in a few of our bundles.  When I click the "bundle.yaml" link from the bundle landing page, I see it has "num_units: 1" in the yaml for the plugin charm.  This is problematic for a couple reasons:  First, the machine count isn't correct (subordinates don't get a new machine).  Second, and here's the biggie, "Add to demo" deployment fails:

> Bundle deployment failed
> Unable to deploy the bundle. The server returned the following error: Invalid number of units [1] for service: plugin

Example landing page and associated bundle.yaml from the link on the right side of the page:
- https://jujucharms.com/u/bigdata-dev/apache-core-batch-processing/
- https://api.jujucharms.com/charmstore/v4/~bigdata-dev/bundle/apache-core-batch-processing-9/archive/bundle.yaml

Note there's a num_units attribute for plugin, even though this is a subordinate charm.  Now take a look at the launchpad source, where there is no such attribute for plugin:
http://bazaar.launchpad.net/~bigdata-dev/charms/bundles/apache-core-batch-processing/trunk/view/head:/bundles.yaml

Interestingly enough, the "bundles.yaml.orig" from the same landing page does match our lp source:
https://api.jujucharms.com/charmstore/v4/~bigdata-dev/bundle/apache-core-batch-processing-9/archive/bundles.yaml.orig

So.. why do we have a "bundle.yaml" and a "bundles.yaml.orig"?  More importantly, can we fix the generated bundle.yaml so it doesn't try to add num_units to subordinate charms?

**Comment by mitechie** [comment]
Thanks for the heads up. This is an issue we've found and a fix is landing as we speak. 

https://github.com/juju/charmrepo/pull/8

The fix until this gets released is to pull down the current yaml file, remove the num_units, and to submit it to bzr with the updated yaml. At that point it'll be using the proper bundle.yaml format and has been proven to be a work around by testing with folks from ecosystems.

**Comment by mitechie** [comment]
Providing the background:

When we moved bundles into the charmstore bundles were updated to be one bundle per file, given updated charmstore urls "cs:mongodb-cluster" and treated more like first class citizens.

Doing this the charmstore ingests the original yaml (v3 api format) and transforms it to the new yaml (v4 format) which introduced the num_units bug as the charmstore auto added a num_units to any service that did not specify.

The .orig is the v3 file and the bundle.yaml (note the lack of an s) is the transformed file.
